### ISSUE TITLE
Crash in type narrowing code

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3351]
The pylance telemetry indicates that a small number of users are seeing a crash in a particular situation. Based on the crash stack crawls, I've identified a repro.

```python
def func():
    c = None

    while True:
        if a:
            if c:
                pass

        a = c == c.foo
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.238, which I just published. It will also be included in the next release of pylance.
